build/tutorials/platform/port-your-ethereum-dapp.md

```diff
@@ -0,0 +1,135 @@
+
+## Developing and deploying contracts
+
+Being an Ethereum-compatible blockchain, all of the usual Ethereum developer tools and environments can be used to develop and deploy dapps for Avalanche's C-Chain. 
+
+### Remix
+
+[Here](../smart-contracts/deploy-a-smart-contract-on-avalanche-using-remix-and-metamask.md) is a tutorial for using Remix to deploy smart contracts on Avalanche. It relies on MetaMask for access to Avalanche network.
+
+### Truffle
+
+You can also use Truffle to test and deploy smart contracts on Avalanche. Find out how in [this](../smart-contracts/using-truffle-with-the-avalanche-c-chain.md) tutorial.
+
+### Hardhat
+
+Hardhat is the newest development and testing environment for Solidity smart contracts, and the one our developers use the most. Due to its superb testing support, it is the recommended way of developing for Avalanche.
+
+[Here](https://github.com/ava-labs/avalanche-smart-contract-quickstart) is a quickstart repository that our developers use to start new projects.
+
+## Avalanche Explorer
+
+Essential part of the smart contract development environment is the explorer, that indexes and serves blockchain data. Main net C-Chain explorer is available at [https://cchain.explorer.avax.network/](https://cchain.explorer.avax.network/) and test net explorer at [https://cchain.explorer.avax-test.network](https://cchain.explorer.avax-test.network/). Besides the web interface, it also exposes the standard [Ethereum JSON RPC API](https://eth.wiki/json-rpc/API).
+
+## Avalanche Faucet
+
+For development purposes, you will need test tokens. Avalanche has a [Faucet](https://faucet.avax-test.network/) that drips tet tokens to address of your choice. Paste your C-Chain address there.
+
+If you need, you can also run a faucet locally, but building it from the [repository](https://github.com/ava-labs/avalanche-faucet).
+
+## Gotchas and things to look out for
+
+Avalanche Platform's C-Chain is EVM-compatible, but it is not identical. There are some differences you need to be aware of, otherwise you may create subtle bugs or inconsistencies in how you dapps behave.
+
+Here are the main differences you should be aware of.
+
+### Measuring time
+
+It is customary on Ethereum to use block height progress as a proxy for time. You should not do that on Avalanche. Chains on Avalanche are quiescent, meaning that if there is no activity, there are no blocks produced. The opposite is also true, if there is great amount of activity, blocks are produced very fast. Because of that, you should not measure the passage of time by the amount of blocks that are produced. You both get inaccurate results, and be oen to manipulation by third parties.
+
+Instead of block rate, you should measure time simply by reading the timestamp attribute of the produced blocks. Timestamps are guaranteed to be increasing, and to be within 30s of the real time.
```

Comment:
they are guaranteed to be monotonically increasing, which means you could have multiple blocks with same timestamp if there is more than 1 block per second, but the timestamp can never go down.

Reply:
added